examples/core/input/0_input_keyboard/main.cpp

```diff
@@ -18,14 +18,22 @@ int main() {
         std::cout << "key press\n";
     };
 
-    auto input = window->get_input();
+    auto input = std::make_shared<Input>();
+    auto input_executor = std::make_shared<InputExecutor>(input);
+
+    KeyListener key_listener = KeyListener(input_executor);
+    MouseListener mouse_listener = MouseListener(input_executor);
```

Comment:
same as abobe

Reply:
That's my code style. I don't think it harms our code in any way. For me, the auto keyword reduces readability. I don't like to use it, except when there is std::shared_pointer.